lib/appsignal/transaction.ex

```diff
@@ -253,7 +253,19 @@ defmodule Appsignal.Transaction do
   @spec set_sample_data(Transaction.t() | nil, String.t(), any) :: Transaction.t()
   def set_sample_data(nil, _key, _payload), do: nil
 
+  def set_sample_data(%Transaction{} = transaction, "params" = key, payload) do
+    if config()[:send_params] do
+      do_set_sample_data(transaction, key, payload)
```

Comment:
What would be the reason we have the `do_set` and the `set` method? Wouldn't only tghe `set_sample_data` be enough?

Reply:
calling `set_sample_data` again would cause it to get trapped in an infinite loop, right?

Reply:
Oh right, didn't see the method overloading.